### ISSUE TITLE
Auromancy

### ISSUE THREAD

**Issue by shoverno2** [TheCodex6824/ThaumicAugmentation#10]
I have seen many addon for thaumcraft 6, but only capable of them adds new auromancy effect and, for now, only this addon adds new effect and STILL ALIVE)))
Why not to add a little bit of auromancy?)
Like an:
1)Effect application
Simple and useful in creative dreams
Also, if this effect will be added, that should have config whith list of effects that is possible to apply/can't be applied
As this function using potion effects usually, this should use alchemy aspect crystal.
2) Negative versions, like:
1]Purification (negative flux)
This should damage flux creatures, purificate block and etc. This might be hard to make. Should use ordo crystal.
2]Blessing (negative curse)
This will not heal, but work as boost, maybe useful in coop multiplayer or for boosting your self. Should use... duno, something contain light, life, kindness.
3] etc))
3) Beam variant
This should be low power high rate variant to attach effect. Maybe easy to make, couse thaumcraft already have something like that in files.
4) Lightning effect
Should be like curse, but making area electrified. Most similar to lightning effect aspect is potentia.
5)Warp(teleport, location swap, watever🙃)
Why not, just why not))
6)Runic sheld recharge
Sometimes player need more runic defence more then his life))
Should use armour aspect crystal.
7)item magnetism
Useful thing, i should say.
Vacous(void) aspect, maybe.
8) Telekinesis
Birds is singing, flowers are blooming...
Yep, mob location manipulation effect.
Hold in air and push away effect should be here.
9)And other crazy and useful thing, that you can imagine))
...
Also, there will be some problems.
First of all, focus bag will be too small. Why not to make advanced version?
Also, sometimes player can't use vis from aura to use focus(vanila bag, dim doors).
Why not to make some battery for those situation?
...
Well, this long))
With best wishes, Dima))))

**Comment by TheCodex6824** [comment]
Thank you for leaving a lot of suggestions! Here's my thoughts on them:

> Effect application
I want to avoid adding "generic" effects that can do everything, so people can't have one super focus. This would also make potions even less relevant than they already are.

> Purification
This sounds like a good addition, perhaps with praecantatio instead of ordo (because flux is praecantatio + perditio). But either way, it makes sense to have a focus effect effective against taint.

> Blessing
I'm not quite sure what you mean by this, but if you mean an area of effect buff effect that's the helping version of curse, I feel like a cloud healing effect would do the same thing.

> Beam
This sounds really cool actually. I'm unsure if Thaumcraft's casting system will let me make continuous use foci without breaking everything, in the event it doesn't I can make it a piercing beam or something like that.

> Lightning
Possibly, but again I feel like the cloud medium would functionally do the same thing.

> Teleport
I don't think I'm going to add this, because it would outclass any other addons' teleportation, as well as partially invalidate the point of the dimension being added in the next update in this addon (with that being very fast travel, similar to the nether)

> Runic shield
I could maybe add this, but it might be a bit too powerful if people have large amounts of runic shielding. It does give me an idea though - it might be worth thinking about a "shield" effect that gives a small amount of runic shielding for a limited duration.

> Item magnetism
The telekinesis effect from Thaumic Tinkerer is able to move items like you described. As long as it is being maintained, I don't want to steal content from them.

> Telekinesis
It would probably have to be called something different to not be confused with what I mentioned above, but this maybe could be added.

> Focus bag
It probably makes sense to add a bigger bag, the only thing I'm worried about is the focus selection menu running out of space or being too cramped to be able to choose foci without frustration.

> Lack of Vis
I've actually never run into a situation where there was 0 vis in any dimension, save for bugs (even DimDoors pockets have vis for me). It might be worth adding a bauble that can provide vis like that if this is a consistent thing, but otherwise I'd be afraid of enabling people to spam high cost effects without consequence.

For the things I think are OK, I want to finish up the update I'm working on now first before adding this stuff, so I can't give an ETA for when they will be implemented. Hopefully soon though :)

**Comment by slowiu06** [comment]
You don't have to worry about Thaumic Tinkereer atleast until you do it on your own. This mod is dead ;/

**Comment by shoverno2** [comment]
Well, a have some thoughts aabout thet)

>Effect aplication
Well, you right, potions will be useless.
I'm just playing whith one cool mod, dynamic shadows. this mod also add effectiveness to blind against mobs. Effect application can't be added, but what about blind. There also same aspect, tenebrae.

>Blessing
Healing cloud will just heal, when bleasing will apply buffs, like an:
1)luck
2)Regeneration
3)Speed
4)Haste
5)(maybe) Strength
Nigative curse whithout health manipulation(regeneration is just effect)

>Teleport
I was mean that teleportation distance will be small.

>Runic Shielding
Well, then you should just add absorption effect apply. Much easier and does same effect. But this defence haven't reganeration.

>Telekinesis
I just thinked that how this will work. And now i think that this should create vacuum, that will attract mobs. Why not to call this now "Vacuum"?

>Lack of vis
I just released that sometimes focuses cant be used in limbo and dungeon dimensions from this mod. Now, well, duno :/

**Comment by shoverno2** [comment]
Also, what about Eldrich power?
That should be like flux element, but less damage and whither effect application.
That will be TAINTED MAGIC))
(Do you know this brilliant mod for thaumcraft 4?)

**Comment by shoverno2** [comment]
Some new ideas

>Cutting wave
Type of attaching effect
Cut through mobs
distance 3 blocks
Wave is pretty big, like sword cut animation now.

>Nova
Did you starting to search for some sort of explosion or other AoE effect, when firstly looked into thaumcraft's 6 auromancy?)
This should have in focus craft config (attack enemy/frends)
(radius 1/2/3 blocks)

For now that's all)

**Comment by shoverno2** [comment]
I wonder, how warding affect player?
I think in fact it's should to make entity immortal, freeze movements and make unable to do anything for a while.
Or it should to... restore runic shield? Or apply little ammount of absorption? 
Or even don't do anything else with this effect)

**Comment by shoverno2** [comment]
Also. Did auramancy targeting system able to be modded? Couse anyone cant chose friends, almost everyone is enemy)

**Comment by KamiSami** [comment]
Homing vis shards, eldritch energy ball (the one Eldritches shoot) and crimson cultist's red lightning energy ball auromancys would be cool, im sure there's plenty more ideas to come by. Auromancy always struck me most interesting part of thaumcraft (probably due to harnessing the power on yourself kinda) along with whole eldritch dimensions and their researches aswell. Looking forward to more auromancy definetily.

**Comment by DoomRater** [comment]
The Thaumonicon said those things are not really Thaumaturgy.  Perhaps they are driven by Impetus in some way?  It's one of those things where I wonder if merely casting such magic would twist the Thaumaturge with every use.

**Comment by TheCodex6824** [comment]
Yeah the eldritch-y attacks are not really suitable lorewise for a focus. There was some ideas of bringing the primal focus back, but the focal manipulator system is not very accommodating of the ideas we had of how to make a focus with that effect, so I'm not sure if it will be done or not.
